2. Add GSN provider support

drortirosh opened the pull request:
minimal set of changes to add GSN support to the contract and the dApp
so that calling user no longer needs eth.

Actual payment is done by the default "accept everything" paymaster
contract.

bhasva commented:
Hi, while I am doing with GSN getting this kind of error
Relaying errors (1):
http://127.0.0.1:65432 => paymaster rejected in local view call to 'relayCall()' : FWD: signature mismatch

drortirosh commented:
> Hi, while I am doing with GSN getting this kind of error Relaying errors (1): http://127.0.0.1:65432 => paymaster rejected in local view call to 'relayCall()' : FWD: signature mismatch

please try to follow the https://docs.opengsn.org/faq, as I think there is such an issue there.
If not, please use our https://forum.opengsn.org to ask more questions, as the "issues" is less appropriate for them.

Tranquil-Flow reviewed (commented):
In CaptureTheFlag.sol, the implementation for setting the trustedForwarder on L15 is no longer correct. Suggest using:

_setTrustedForwarder(forwarder);

drortirosh committed:
Add GSN provider support

minimal set of changes to add GSN support to the contract and the dApp
so that calling user no longer needs eth.

Actual payment is done by the default "accept everything" paymaster
contract.
